dbus-cxx/sendmsgtransport.cpp

```diff
@@ -342,6 +348,7 @@ ssize_t SendmsgTransport::writeMessage( std::shared_ptr<const DBus::Message> mes
 }
 
 std::shared_ptr<DBus::Message> SendmsgTransport::readMessage() {
+    std::scoped_lock lock(m_priv->m_receive_mutex);
```

Comment:
this should not be needed, because only the dispatcher thread should be reading messages.  Is there some situation that you found where that was not the case?